packages/addon-dev/src/rollup-gjs-plugin.ts

```diff
@@ -0,0 +1,71 @@
+import { createFilter } from '@rollup/pluginutils';
+import type { Plugin, PluginContext, ResolvedId } from 'rollup';
+import { readFileSync } from 'fs';
+import { Preprocessor } from 'content-tag';
+
+const PLUGIN_NAME = 'rollup-gjs-plugin';
+
+const processor = new Preprocessor();
+// import { parse as pathParse } from 'path';
+
+export default function rollupGjsPlugin(): Plugin {
+  return {
+    name: PLUGIN_NAME,
+    async resolveId(source: string, importer: string | undefined, options) {
+      let resolution = await this.resolve(source, importer, {
+        skipSelf: true,
+        ...options,
+      });
+
+      if (resolution) {
+        return maybeRewriteGJS(resolution);
+      }
+    },
+
+    load(id: string) {
+      const meta = getMeta(this, id);
+      if (!meta) {
+        return;
+      }
+
+      this.addWatchFile(meta.originalId);
+      let input = readFileSync(meta.originalId, 'utf8');
+      let code = processor.process(input);
+      return {
+        code,
+      };
+    },
+  };
+}
+
+type Meta = {
+  originalId: string;
+};
+
+function getMeta(context: PluginContext, id: string): Meta | null {
+  const meta = context.getModuleInfo(id)?.meta?.[PLUGIN_NAME];
+  if (meta) {
+    return meta as Meta;
+  } else {
+    return null;
+  }
+}
+
+const gjsFilter = createFilter('**/*.gjs');
```

Comment:
I think if you match gts here too (and rename gts to ts the same way you rename gjs to js) it will all just compose correctly for the typescript case too. content-tag already supports both automatically.

Reply:
so I figured it would be best to transform .gjs to .js files and .gts to .ts files so I expanded the filter and added a bit of code to rename the id correctly

I wanted to have a test but we actually don't have the TS setup correctly on the addon and rollup is complaining 🙈